### PR TITLE
Fix for #13740

### LINKED ISSUE #13740: [MLB]  ERROR: Unable to extract video id
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like that [x])
- Use *Preview* tab to see how your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2017.07.23*. If it's not read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [X] I've **verified** and **I assure** that I'm running youtube-dl **2017.07.23**

### Before submitting an *issue* make sure you have:
- [X] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [X] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [X] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add `-v` flag to **your command line** you run youtube-dl with, copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):
```
youtube-dl.py -v "https://www.mlb.com/video/hargrove-homers-off-caldwell/c-1352023483?tid=67793694"
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://www.mlb.com/video/hargrove-homers-off-caldwell/c-1352023483?tid=67793694']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2017.07.23
[debug] Python version 3.5.1 - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg N-71727-g46778ab, ffprobe 3.2, rtmpdump 2.4
[debug] Proxy map: {}
[MLB] c-1352023483?tid=67793694: Downloading webpage
ERROR: Unable to extract video id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the -
-verbose flag and include its complete output.
Traceback (most recent call last):
  File "C:\Transmogrifier\youtube-dl.py\youtube_dl\YoutubeDL.py", line 776, in extract_info
    ie_result = ie.extract(url)
  File "C:\Transmogrifier\youtube-dl.py\youtube_dl\extractor\common.py", line 433, in extract
    ie_result = self._real_extract(url)
  File "C:\Transmogrifier\youtube-dl.py\youtube_dl\extractor\mlb.py", line 132, in _real_extract
    [r'data-video-?id="(\d+)"', r'content_id=(\d+)'], webpage, 'video id')
  File "C:\Transmogrifier\youtube-dl.py\youtube_dl\extractor\common.py", line 782, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract video id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure
 to call youtube-dl with the --verbose flag and include its complete output.

```

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):

- Single video: https://www.mlb.com/video/hargrove-homers-off-caldwell/c-1352023483?tid=67793694

Thanks
Ringo

### PR DESCRIPTION
**Pull request by mmcru**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information
Adjust `_VALID_URL` in mlb.py to fix #13740

### CODE REVIEW

**Comment by dstftw** [comment]
Add a test.

**Comment by mmcru** [comment]
Done :)

File: youtube_dl/extractor/mlb.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ class MLBIE(InfoExtractor):
                         (?:[\da-z_-]+\.)*mlb\.com/
                         (?:
                             (?:
-                                (?:.*?/)?video/(?:topic/[\da-z_-]+/)?v|
+                                (?:.*?/)?video/(?:topic/[\da-z_-]+/)?(?:v|(?:.*?/)c-)|
```
Comment (dstftw) [inline youtube_dl/extractor/mlb.py]: Inner group is pointless.

File: youtube_dl/extractor/mlb.py [hunk 2]
```diff
@@ -95,6 +95,18 @@ class MLBIE(InfoExtractor):
...
+            'url': 'https://www.mlb.com/video/hargrove-homers-off-caldwell/c-1352023483?tid=67793694',
+            'md5': 'b2af6c81fcd6c8edd9c74daea0676359',
+            'info_dict': {
+                'id': '1352023483',
+                'ext': 'mp4',
+                'title': 'Hargrove homers off Caldwell',
+                'description': '7/31/82: Mike Hargrove shows off a lengthy batting routine, then homers off Mike Caldwell',
+                'timestamp': 396934380,
+                'upload_date': '19820731',
+            }
```
Comment (dstftw) [inline youtube_dl/extractor/mlb.py]: Make test `only_matching`.

**Pull request merged**